sched/queue/memory/simple.go

```diff
@@ -48,33 +142,28 @@ func (q *simpleQueue) Enqueue(job sched.Job) (string, error) {
 		return "", err
 	}
 	resultCh := make(chan enqueueResult)
-	q.inCh <- enqueueReq{job, resultCh}
+	q.reqCh <- enqueueReq{job, resultCh}
 	result := <-resultCh
 	return result.jobID, result.err
 }
 
-func (q *simpleQueue) Close() error {
-	close(q.inCh)
-	close(q.outCh)
-	return nil
+type statusResult struct {
+	status sched.JobStatus
+	err    error
 }
 
-func (q *simpleQueue) loop() {
-	for {
-		req, ok := <-q.inCh
-		if !ok {
-			// Channel is cloed
-			return
-		}
-		job := req.job
-		id := fmt.Sprintf("%v", q.nextID)
-		q.nextID++
-		job.Id = id
-		select {
-		case q.outCh <- simpleWorkItem(job):
-			req.resultCh <- enqueueResult{id, nil}
-		default:
-			req.resultCh <- enqueueResult{"", queue.NewCanNotScheduleNow(1*time.Second, "queue full")}
-		}
-	}
+func (q *simpleQueue) Status(jobId string) (sched.JobStatus, error) {
+	resultCh := make(chan statusResult)
+	q.reqCh <- statusReq{jobId, resultCh}
+	result := <-resultCh
```

Comment:
why is the status request added to the queue?  You have enough information in the queueState to answer this immediately.  Either its already been queued (PENDING) or its not (UNKNOWN)

What benefit is their of the increased latency for this read being answered in order of requests being processed?